ocs_ci/deployment/disconnected.py

```diff
@@ -100,6 +102,28 @@ def mirror_images_from_mapping_file(mapping_file, icsp=None, ignore_image=None):
         with open(mapping_file, "w") as f:
             f.writelines(mapping_file_content)
 
+    if idms:
+        # update mapping.txt file with urls updated based on provided
+        # ImageDigestMirrorSet
+        with open(mapping_file) as mf:
+            mapping_file_content = []
+            for line in mf:
+                # exclude ignore_image
+                if ignore_image and ignore_image in line:
+                    continue
+                # apply any matching policy to all lines from mapping file
+                for policy in idms["spec"]["imageDigestMirrors"]:
+                    # we use only first defined mirror for particular source,
+                    # because we don't use any IDMS with more mirrors for one
+                    # source and it will make the logic very complex and
+                    # confusing
+                    line = line.replace(policy["source"], policy["mirrors"][0])
+                mapping_file_content.append(line)
+        # write mapping file to disk
+        mapping_file = "_updated".join(os.path.splitext(mapping_file))
+        with open(mapping_file, "w") as f:
+            f.writelines(mapping_file_content)
+
```

Comment:
Here I would propose to do common `if icsp or idms:` section and than do the differentiation for icsp/idms only for the on the minimal part of the code where needed, so the common parts of code will not be duplicated.
For example instead of:
```python
for policy in idms["spec"]["imageDigestMirrors"]:
```
I would propose something like:
```python
policies = idms["spec"]["imageDigestMirrors"] if idms else icsp["spec"]["repositoryDigestMirrors"]
for policy in policies:
    ...
```

Reply:
anyway we would be removing icsp thus I created a seperate section as it would be easy to remove, but this also looks clean will update it like wise